### ISSUE TITLE
The hope is to support internal streaming returns in the agent, with the utility functions returning non-empty values.

### ISSUE THREAD

**Issue by zhangwei668** [microsoft/autogen#4825]
### What feature would you like to be added?

I want to inherit the `agent` module and rewrite `create` as `create_stream`. However, this causes issues with tool usage, resulting in empty returns. The reason for this is that the `create` and `create_stream` methods in the `OpenAIChatCompleteClient` class are written inconsistently.

### Why is this needed?

The hope is to support internal streaming returns in the agent, with the utility functions returning non-empty values.

**Comment by ekzhu** [comment]
We are working on streaming support in an existing issue.

**Comment by KnSup** [comment]
I'm confused about the issue mentioned in #3862. What I currently need is the effect of using `await self._model_client.create_stream` in `on_messages_stream`. However, it seems that `await self._model_client.create` is being used now. This results in information being returned to the user only after each agent has finished running, which is not a good user experience.

**Comment by ekzhu** [comment]
The issue mentioned is exactly to use the `run_stream` API like `create_stream`. Could you let us know what is the confusion?

**Comment by KnSup** [comment]
`run_stream` allows me to obtain the information after each Agent has finished executing, but it doesn't provide the streaming output content like `create_stream` does.

**Comment by ekzhu** [comment]
You are right -- this is exactly what we are working on in the issue mentioned above.
